clients/p4rt_perf_test/p4rt_perf_main.cc

```diff
@@ -0,0 +1,201 @@
+ * focus is on measuring the time required to program a specified number of
+ * entries.
+ *
+ * Command Line Arguments:
+ *   -t : Number of threads (optional, default: 1).
+ *   -o : Operation type (mandatory). 1 is to ADD and 2 to DEL
+ *   -n : Number of entries (optional, default: 1,000,000).
+ *   -p : P4 profile (optional, default: SIMPLE_L2_DEMO).
+ *
+ * The application evaluates the time taken to perform the specified operation
+ * on the defined number of entries within the chosen P4 profile. It supports
+ * multithreading to simulate concurrent operations and can be configured via
+ * command line arguments.
+ *
+ * TODO:
+ * 1. Programming with multiple threads doesn't really work as stratum currently
+ * supports write by only master.
+ * 2. Logging
+ *
+ */
+
+#include <arpa/inet.h>
+#include <pthread.h>
+
+#include <chrono>
+#include <ctime>
+#include <iostream>
+#include <thread>
+
+#include "absl/flags/flag.h"
+#include "absl/memory/memory.h"
+#include "p4rt_perf_session.h"
+#include "p4rt_perf_simple_l2_demo.h"
+#include "p4rt_perf_test.h"
+#include "p4rt_perf_tls_credentials.h"
+
+#define MAX_THREADS 8
+#define CORE_BASE 20
+
+// globals
+TestParams test_params = {};
+ThreadInfo thread_data[MAX_THREADS];
+uint8_t core_id[MAX_THREADS];
+
+ABSL_FLAG(std::string, grpc_addr, "localhost:9559",
+          "P4Runtime server address.");
+ABSL_FLAG(uint64_t, device_id, 1, "P4Runtime device ID.");
+
+using P4rtStream = ::grpc::ClientReaderWriter<p4::v1::StreamMessageRequest,
+                                              p4::v1::StreamMessageResponse>;
+
+void PopulateThreadInfo() {
+  uint16_t index;
+  uint64_t num_entries_thread =
+      test_params.tot_num_entries / test_params.num_threads;
+  uint64_t rem_rules = test_params.tot_num_entries % test_params.num_threads;
+
+  for (index = 0; index < test_params.num_threads; index++) {
+    thread_data[index].tid = index;
+    thread_data[index].start = index * num_entries_thread;
+    thread_data[index].num_entries = num_entries_thread;
+    thread_data[index].oper = test_params.oper;
+
+    /* Initialize Core Ids */
+    core_id[index] = CORE_BASE + index;
+    thread_data[index].core_id = core_id[index];
+  }
+
+  // add remaining entries to the last thread;
+  thread_data[test_params.num_threads - 1].num_entries += rem_rules;
+
+  // dump the data
+  for (index = 0; index < test_params.num_threads; index++) {
+    printf("Thread data - Core: %d start_index: %d num_entries: %d\n",
+           core_id[index], thread_data[index].start,
+           thread_data[index].num_entries);
+  }
+}
+
+void RunPerfTest(int tid) {
+  // Start a new client session.
+  auto status_or_session = P4rtSession::Create(absl::GetFlag(FLAGS_grpc_addr),
+                                               GenerateClientCredentials(),
+                                               absl::GetFlag(FLAGS_device_id));
+  if (!status_or_session.ok()) {
```

Comment:
Added error message. 
Can't return the status as this is a thread function. So updated the status in thread_data structure which is then checked in the main thread and returned as the status to the user. While this can be done better using 'future' and 'promise' features, I would think this is good enough for a standalone test application.